pymeasure/instruments/common_base.py

```
@@ -166,9 +206,9 @@ class Extreme5000(Instrument):
         :param \\**kwargs: Keyword arguments for all children.
         """
 
-        def __init__(self, cls, id=None, **kwargs):
-            super().__init__(cls=cls, **kwargs)
-            if (isinstance(id, (str, int)) or id is None) and self.valid_class:
+        def __init__(self, cls: type[Child], id: IdType = None, **kwargs):
```

Ruff (0.15.12)

[error] Function argument `id` is shadowing a Python builtin

(A002)